wefe/debias/half_sibling_regression.py

```diff
@@ -0,0 +1,351 @@
+     | and Jonas Peters. Modeling confounding by half-sibling regression.
+     | Proceedings of the National Academy of Sciences, 113(27):7391–7398, 2016
+
+    """
+
+    name = "Half Sibling Regression"
+    short_name = "HSR"
+
+    def __init__(
+        self,
+        verbose: bool = False,
+        criterion_name: Optional[str] = None,
+    ) -> None:
+        """Initialize a Half Sibling Regression Debias instance.
+
+        Parameters
+        ----------
+        verbose : bool, optional
+            True will print informative messages about the debiasing process,
+            by default False.
+        criterion_name : Optional[str], optional
+            The name of the criterion for which the debias is being executed,
+            e.g., 'Gender'. This will indicate the name of the model returning
+            transform, by default None
+        """
+        # check verbose
+        if not isinstance(verbose, bool):
+            raise TypeError(f"verbose should be a bool, got {verbose}.")
+
+        self.verbose = verbose
+
+        if criterion_name is None or isinstance(criterion_name, str):
+            self.criterion_name_ = criterion_name
+        else:
+            raise ValueError(f"criterion_name should be str, got: {criterion_name}")
+
+    def _get_gender_vectors(
+        self, model: WordEmbeddingModel, gender_definition: List[str]
+    ) -> np.ndarray:
+
+        vectors = [model[word] for word in gender_definition if word in model]
+        return np.asarray(vectors)
+
+    def _get_non_gender_dict(
+        self, model: WordEmbeddingModel, non_gender: List[str]
+    ) -> Dict[str, float]:
+
+        dictionary = get_embeddings_from_sets(
+            model=model, sets=[non_gender], sets_name="non_gender", normalize=False
+        )
+        return dictionary[0]
+
+    def _compute_weigth_matrix(
+        self, gender_vectors: np.ndarray, non_gender_vectors: np.ndarray, alpha: float
+    ) -> np.ndarray:
+
+        a = gender_vectors.T @ gender_vectors + alpha * np.eye(gender_vectors.shape[1])
+        b = gender_vectors.T @ non_gender_vectors
+        weight_matrix = np.linalg.inv(a) @ b
+        return weight_matrix
+
+    def _compute_gender_information(
+        self, gender_vectors: np.ndarray, weight_matrix: np.ndarray
+    ) -> np.ndarray:
+        gender_information = gender_vectors @ weight_matrix
+        return gender_information
+
+    def _substract_gender_information(
+        self, non_gender_vectors: np.ndarray, gender_information: np.ndarray
+    ) -> np.ndarray:
+        debiased_vectors = non_gender_vectors - gender_information
+        return debiased_vectors
+
+    def _get_indexes(
+        self, model, target: List[str], non_gender: List[str]
+    ) -> List[int]:
+        return [non_gender.index(word) for word in target if word in model]
+
+    def fit(
+        self,
+        model: WordEmbeddingModel,
+        gender_definition: Sequence[str],
```

Comment:
Here one could generalize the name to something like:

```suggestion
        bias_definitional_words: Sequence[str],
```

Reply:
yes, now gender has been generalized to "bias" in the method